lib/packwerk/configuration.rb

```diff
@@ -58,10 +58,13 @@ def all_application_autoload_paths
         (engine.config.autoload_paths + engine.config.eager_load_paths + engine.config.autoload_once_paths).uniq
       end
 
+      rails_root_match = Rails.root.join("**").to_s
+      bundle_path_match = Bundler.bundle_path.join("**").to_s
+
       all_paths = all_paths.map do |path_string|
-        # ignore paths outside of the Rails root
+        # ignore paths outside of the Rails root and gems
         path = Pathname.new(path_string)
-        if path.exist? && path.realpath.fnmatch(Rails.root.join("**").to_s)
+        if path.exist? && path.realpath.fnmatch(rails_root_match) && !path.realpath.fnmatch(bundle_path_match)
```

Comment:
Heh. We introduced the restriction to paths under `Rails.root` mostly to eliminate gem load paths, but now we have a separate clause for gem load paths.

In this case do we even still need the `rails_root_match`?

Reply:
Good point. I can't think of anything other than gems which could add themselves to the load paths, so simply ignoring the `bundle_path` might be sufficient.

Reply:
I have tried this out on a large project which vendors gems. You are correct that now we are checking the bundle path, there is no need for the separate check of `Rails.root`. I have added a new commit with that change.

I have also added the `vendor` directory to the default `exclude` list. I can't think of a reason you'd want Packwerk to scan files inside the gems.